clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/CoreNotificationRenderer.java

```diff
@@ -141,4 +138,40 @@ public String getActionButtonIconKey() {
         return Constants.NOTIF_ICON;
     }
 
+    @Override
+    public Builder setSound(final Context context, final Bundle extras, final Builder nb,CleverTapInstanceConfig config
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.